tests/mpi_54rank/test_external_grid_1x1.py

```diff
@@ -0,0 +1,154 @@
+        rank=cube_comm.rank,
+        global_dims=[pace.util.Y_INTERFACE_DIM, pace.util.X_DIM],
+        global_extent=(npy, nx),
+        overlap=True,
+    )
+
+    subtile_slice_dy = cube_comm.partitioner.tile.subtile_slice(
+        rank=cube_comm.rank,
+        global_dims=[pace.util.Y_DIM, pace.util.X_INTERFACE_DIM],
+        global_extent=(ny, npx),
+        overlap=True,
+    )
+
+    subtile_slice_area = cube_comm.partitioner.tile.subtile_slice(
+        rank=cube_comm.rank,
+        global_dims=[pace.util.Y_DIM, pace.util.X_DIM],
+        global_extent=(ny, nx),
+        overlap=True,
+    )
+
+    lon_rad = lon * (PI / 180)
+    lat_rad = lat * (PI / 180)
+
+    errors = []
+    diffs = []
+
+    if not np.isclose(
+        ext_driver.state.grid_data.lon.view[:, :], lon_rad[subtile_slice_grid]
+    ).all():
+        errors.append("Lon data mismatch")
+
+    diff_lon = np.amax(
+        ext_driver.state.grid_data.lon.view[:, :] - lon_rad[subtile_slice_grid]
+    ) / np.amax(lon_rad[subtile_slice_grid])
+    diffs.append(f"Lon maximum relative error = {diff_lon}")
+
+    if not np.isclose(
+        ext_driver.state.grid_data.lat.view[:, :], lat_rad[subtile_slice_grid]
+    ).all():
+        errors.append("Lat data mismatch")
+
+    diff_lat = np.amax(
+        ext_driver.state.grid_data.lat.view[:, :] - lat_rad[subtile_slice_grid]
+    ) / np.amax(lat_rad[subtile_slice_grid])
+    diffs.append(f"Lat maximum relative error = {diff_lat}")
+
+    if not np.isclose(
+        ext_driver.state.grid_data.dy.view[:, :], dx[subtile_slice_dx]
+    ).all():
+        errors.append("dx data mismatch")
+
+    diff_dx = np.amax(
+        ext_driver.state.grid_data.dy.view[:, :] - dx[subtile_slice_dx]
+    ) / np.amax(dx[subtile_slice_dx])
+    diffs.append(f"dx maximum relative error = {diff_dx}")
+
+    if not np.isclose(
+        ext_driver.state.grid_data.dx.view[:, :], dy[subtile_slice_dy]
+    ).all():
+        errors.append("dy data mismatch")
+
+    diff_dy = np.amax(
+        ext_driver.state.grid_data.dx.view[:, :] - dy[subtile_slice_dy]
+    ) / np.amax(dy[subtile_slice_dy])
+    diffs.append(f"dy maximum relative error = {diff_dy}")
+
+    if not np.isclose(
+        ext_driver.state.grid_data.area.view[:, :], area[subtile_slice_area]
+    ).all():
+        errors.append("area data mismatch")
+
+    diff_area = np.amax(
+        ext_driver.state.grid_data.area.view[:, :] - area[subtile_slice_area]
+    ) / np.amax(area[subtile_slice_area])
+    diffs.append(f"Area maximum relative error = {diff_area}")
+
+    assert not errors, "errors occured in 1x1:\n{}".format("\n".join(errors))
+
+    surface_area_true = 4 * PI * (RADIUS ** 2)
+
+    mpicomm = MPIComm()
+
+    tmp = [math.fsum(row) for row in ext_driver.state.grid_data.area.view[:, :]]
+    rank_area_sum = math.fsum(tmp)
+    tile_area = mpicomm._comm.gather(rank_area_sum, root=0)
+
+    if mpicomm.Get_rank() == 0:
+        total_area = math.fsum(tile_area)
+        print(diffs)
```

Comment:
does this mean you're only printing the diffs on rank 0?

Reply:
Ah yes, rank 0 would be the only one printing its diffs, I will change this so that all ranks print their relative errors.